client/cli/src/params/shared_params.rs

```diff
@@ -22,7 +22,7 @@ use crate::arg_enums::Database;
 /// Shared parameters used by all `CoreParams`.
 #[derive(Debug, StructOpt, Clone)]
 pub struct SharedParams {
-	/// Specify the chain specification (one of dev, local, or staging).
+	/// Specify the chain specification (one of <chain_name>, dev, <chain_name>-dev, <chain_name>-local, or <chain_name>-staging where <chain_name> is polkadot, kusama, westend, etc...)
```

Comment:
```suggestion
	/// Specify the chain specification.
```

That is the only correct documentation here. We should add a CLI subcommand to list the supported chain specs, but these are different for each project and the respective names.

Reply:
That would be ideal indeed. Until we have this new CLI, don't you think having a doc currently matching would be ok ?

Reply:
No. Substrate is not only used by Polkadot ;)

Reply:
I know, thus the `etc...` :)

Reply:
Still not convinced. If something isn't clear, we should update the Polkadot Readme at the moment.

Reply:
(It shouldn't be complicated to make a subcommand though)

Reply:
I mean, just add a variant in client/cli/src/commands/mod.rs, add a new function in the trait SubstrateCli (client/cli/src/lib.rs) and call that function and display when the command is ran.

So every app can defined this function that will be just next to the load_spec of SubstrateCli

Reply:
I agree and I will love to do it but I don't have the time right now. This is why I provided a 'quick' fix so users who do RTFM get at least on the right track.

I agree 100% about the benefit of the option from @bkchr

Reply:
So, to create a subcommand you need to:

1. Add a variant in [this enum](https://github.com/paritytech/substrate/blob/master/client/cli/src/commands/mod.rs#L40-L59)

2. Add a struct command (the inner struct of the enum variant) to its own file. Example [here](https://github.com/paritytech/substrate/blob/master/client/cli/src/commands/check_block_cmd.rs#L30-L48) with CheckBlockCmd.

3. Make a method "run" for this struct. It doesn't need to be async but you *can* if you would need to.

    It probably doesn't need any parameter (not even config I suppose).

    It does requires a type parameter for the Cli. So you can write: `fn run<C: SubstrateCli>()`

4. Add a match arm in the runner to run the command [here](https://github.com/paritytech/substrate/blob/master/client/cli/src/runner.rs#L176-L189). You can use `cmd.run(...)` if the method is sync or use the `run_until_exit` if the method is async.

    You will need to pass the type parameter. Here it's given by the runner itself which is C: `cmd.run::<C>()` (no argument is used here as I explain in point 3.).

5. Now to actually implement the feature, we want the application to define itself the function that will return the accepted values (not substrate). To do so, we are going to do it on the application-level trait: SubstrateCli that you can find [here](https://github.com/paritytech/substrate/blob/master/client/cli/src/lib.rs#L57-L191).

    In this trait you can add a method without default implementation `fn available_chains() -> Vec<String>;`. Because this method is defined in the trait, every application will have to provide an implementation by themselves, thus making the compilation of node and node-template fail because they will need to define this method on their SubstrateCli.

    But first, let's assume the applications have provided an implementation. Go to your `fn run()` method you have implemented in the command struct you have created and println the list in a fancy way:

    ```rust
    for chain in C::available_chains() {
        println!(" - {}", chain);
    }
    ```

6. Lastly, you need to add the implementation for the new `available_chains` method in every application.

    a. go in node-template, command.rs: add one that returns the possible string
    b. go to bin/node, command.rs: add one that returns the possible string
    c. create a polkadot branch and a PR, go to cli/src/commands.rs and add the implementation there too

    Example:
    ```rust
    fn available_chains() -> Vec<String> {
        vec!["dev", "local", "<file>"]
    }
    ```

7. (Well... we will see at the review but there will probably be a need to optimize things up because we could return a static array of static str instead of a Vec newly created. But try to make it works first with a simple Vec of String.)